openedx/core/djangoapps/catalog/management/commands/tests/test_cache_programs.py

```diff
@@ -39,10 +42,27 @@ def setUp(self):
         self.list_url = self.catalog_integration.get_internal_api_url().rstrip('/') + '/programs/'
         self.detail_tpl = self.list_url.rstrip('/') + '/{uuid}/'
         self.pathway_url = self.catalog_integration.get_internal_api_url().rstrip('/') + '/pathways/'
+        self.course_run_url = self.catalog_integration.get_internal_api_url().rstrip('/') + '/course_runs/'
 
         self.programs = ProgramFactory.create_batch(3)
         self.pathways = PathwayFactory.create_batch(3)
 
+        fake_course_runs = {}
+        for program in self.programs:
+            for course in program['courses']:
+                for course_run in course['course_runs']:
+                    course_run_key = course_run['key']
+                    course_run_already_in_list = fake_course_runs.get(course_run_key)
+                    if course_run_already_in_list:
+                        fake_course_runs[course_run_key] += course_run_key
+                    else:
+                        fake_course_runs[course_run_key] = [program['uuid']]
+        self.course_runs = [
+            {'key':cr_key,'programs':[{'uuid':p} for p in programs]} 
+            for cr_key, programs 
+            in fake_course_runs.items()
+        ]
```

Comment:
What is the difference between ```self.course_runs``` and ```fake_course_runs```?  Is ```fake_course_runs``` used later? I think you can replace ```fake_course_runs``` with ```self.course_runs```.